src/NServiceBus.Core/Encryption/RijndaelEncryptionService.cs

```diff
@@ -198,6 +199,13 @@ static bool IsValidKey(byte[] key)
             using (var rijndael = new RijndaelManaged())
             {
                 var bitLength = key.Length * 8;
+
+                var maxValidKeyBitLength = rijndael.LegalKeySizes.OrderByDescending(keyLength => keyLength.MaxSize).Select(keyLength => keyLength.MaxSize).First();
```

Comment:
Because that's what happens when you don't look at your stuff properly.

Updated